### PR TITLE
Add options to addRasterImage

### PR DESCRIPTION
**Pull request by jcheng5**
The other layer methods have `options` parameters, not sure why `addRasterImage` didn't. Apparently @tim-salabim needs this for pane support. https://twitter.com/TimSalabim3/status/1279105173378531335

(Note: Lots of .Rd files changed, this is because I used a more recent roxygen version)

```r
library(raster)

r <- raster(xmn = -2.8, xmx = -2.79, ymn = 54.04, ymx = 54.05, nrows = 30, ncols = 30)
values(r) <- matrix(1:900, nrow(r), ncol(r), byrow = TRUE)
crs(r) <- CRS("+init=epsg:4326")

leaflet() %>% addTiles() %>%
  addMapPane("hello", 100000) %>%
  addRasterImage(r, colors = "Spectral", opacity = 0.8, options = tileOptions(pane = "hello"))
```

PR task list:
- [x] Update NEWS
- [ ] Add tests (where appropriate)
  - R code tests: `tests/testthat/`
  - Visual tests: `R/zzz_viztest.R`
- [x] Update documentation with `devtools::document()`

### CODE REVIEW

File: R/layers.R [hunk 1]
```diff
@@ -243,10 +245,14 @@ addRasterImage <- function(
   project = TRUE,
   method = c("auto", "bilinear", "ngb"),
   maxBytes = 4 * 1024 * 1024,
+  options = tileOptions(),
```
Comment (schloerke) [inline R/layers.R:248]:
```suggestion
  options = gridOptions(),
```

**Comment by tim-salabim** [comment]
Thanks for the quick action!

I can confirm that this works, although, as @schloerke points out, we currently need to use `tileOptions()` to pass additional options.

```r
library(raster)
library(leaflet)
library(leaflet.extras2)

r1 <- raster(xmn = -2.8, xmx = -2.79, ymn = 54.04, ymx = 54.05, nrows = 30, ncols = 30)
values(r1) <- matrix(1:900, nrow(r1), ncol(r1), byrow = TRUE)
crs(r1) <- CRS("+init=epsg:4326")

r2 <- raster(xmn = -2.8, xmx = -2.79, ymn = 54.04, ymx = 54.05, nrows = 30, ncols = 30)
values(r2) <- matrix(1:900, nrow(r2), ncol(r2), byrow = FALSE)
crs(r2) <- CRS("+init=epsg:4326")

leaflet(
  options = leafletOptions(maxZoom = 20)
) %>%
  addMapPane("left", zIndex = 1) %>%
  addMapPane("right", zIndex = 1) %>%
  addTiles(
    group = "base"
    , layerId = "baseid"
    , options = tileOptions(pane = "right")
  ) %>%
  addProviderTiles(
    providers$CartoDB.DarkMatter
    , group = "carto"
    , layerId = "cartoid"
    , options = tileOptions(pane = "left")
  ) %>%
  addRasterImage(
    r1
    , group = "leftImg"
    , opacity = 0.6
    , options = tileOptions(
      , pane = "left"
      , maxZoom = 20
    )
  ) %>%
  addRasterImage(
    r2
    , group = "rightImg"
    , opacity = 1
    , options = tileOptions(
      , pane = "right"
      , maxZoom = 20
    )
  ) %>%
  addLayersControl(overlayGroups = c("leftImg", "rightImg")) %>%
  addSidebyside(
    layerId = "sidecontrols"
    , rightId = "baseid"
    , leftId = "cartoid"
  )
```
Screenshot from 2020-07-04 10-43-56

FWIW, there have been PRs in the past for this:

https://github.com/rstudio/leaflet/pull/508 & https://github.com/rstudio/leaflet/pull/524

Not sure whether it's better to have a dedicated `gridOptions()` or if `tileOptions()` can be used instead.?

**Comment by jcheng5** [comment]
Shoot, I didn't see the past PRs. :man_facepalming:

I created `gridOptions` because `tileOptions` contains some options that don't apply for grid layers. Maybe that's not worth introducing another function for, though?

**Comment by tim-salabim** [comment]
I think `gridOptions` makes things clearer. Tiles are usually something else than a regular grid/raster. So I vote for `gridOptions`.

**Comment by tim-salabim** [comment]
Upon further investigating the [GridLayer](https://leafletjs.com/reference-1.6.0.html#gridlayer) and [TileLayer](https://leafletjs.com/reference-1.6.0.html#tilelayer) options, we should definitely have `gridOptions`. TileLayer inherits options from GridLayer, so it only seems logical to have `gridOptions`.

**Comment by jcheng5** [comment]
Rebase to master before merging

**Comment by rhijmans** [comment]
I would also love to have gridOptions.

**Comment by roelmay** [comment]
Hi, Is there any update on the status on merging the raster options ("joe/feature/raster-options") for an earlier version of leaflet into the main and newest version of leaflet?

**Comment by rhijmans** [comment]
It would be great to have gridOptions

**Comment by amb26** [comment]
I would also love to see the options argument added

**Comment by jeremyash** [comment]
Just re-iterating that options for `addRasterImage` would be great to have. I've been using 'joe/feature/raster-options' to have control over plotting order of raster images with panes, but I'm now also using terra and it's not compatible with those earlier functions. Thanks!

**Comment by jcheng5** [comment]
@jeremyash Thanks for the nudge. I've now updated the `joe/feature/raster-options` branch, but something's completely borked with terra/raster on my travel computer, so I'm not actually able to test it. If you (or anyone reading this) could try it and confirm that it works with both terra and raster objects, I will merge this right away.

**Comment by jeremyash** [comment]
@jcheng5 Just ran some code using terra and raster objects, and it worked fine on my end. Really appreciate your responsiveness here! Many thanks!

**Pull request merged**